backend/geonature/core/auth/routes.py

```diff
@@ -112,6 +112,15 @@ def loginCas():
                 "id_organisme": organism_id,
             }
             response.set_cookie("current_user", str(current_user), expires=cookie_exp)
+
+            # Log the user in
+            user = db.session.execute(
+                sa.select(models.User)
+                .where(models.User.identifiant == current_user["user_login"])
+                .where(models.User.filter_by_app())
+            ).scalar_one()
+            login_user(user)
```

Comment:
**Suggestion:** It's a good practice to validate the user data before using it. Consider adding a check to ensure that `current_user["user_login"]` is not empty or null before using it in the `where` clause. [best practice]
```suggestion
            if current_user["user_login"]:
                user = db.session.execute(
                    sa.select(models.User)
                    .where(models.User.identifiant == current_user["user_login"])
                    .where(models.User.filter_by_app())
                ).scalar_one()
                login_user(user)
            else:
                log.error("User login is empty")
                raise ValueError("User login is empty")
```